util/find-doc-nits

```diff
@@ -203,6 +205,153 @@ sub check_head_style {
     }
 }
 
+# Because we have options and symbols with extra markup, we need
+# to take that into account, so we need a regexp that extracts
+# markup chunks, including recursive markup.
+# please read up on /(?R)/ in perlre(1)
+# (note: order is important, (?R) needs to come before .)
+# (note: non-greedy is important, or something like 'B<foo> and B<bar>'
+# will be captured as one item)
+my $markup_re =
+    qr/(                        # Capture group
+           [BIL]<               # The start of what we recurse on
+           (?:(?-1)|.)*?        # recurse the whole regexp (refering to
+                                # the last opened capture group, i.e. the
+                                # start of this regexp), or pick next
+                                # character.  Do NOT be greedy!
+           >                    # The end of what we recurse on
+       )/x;                     # (the x allows this sort of split up regexp)
+
+# Options must start with a dash, followed by a letter, possibly
+# followed by letters, digits, dashes and underscores, and the last
+# character must be a letter or a digit.
+# We do also accept the single -? or -n, where n is a digit
+my $option_re =
+    qr/(?:
+            \?                  # Single question mark
+            |
+            \d                  # Single digit
+            |
+            [[:alpha:]](?:[-_[:alnum:]]*?[[:alnum:]])?
+       )/x;
+
+# Checks of command option (man1) formatting.  The man1 checks are
+# restricted to the SYNOPSIS and OPTIONS sections, the rest is too
+# free form, we simply cannot be too strict there.
+
+sub is_option_normalise {
+    my ($id, $filename, $option) = @_;
+
+    # $unwrapped is the option with the outer B<> markup removed
+    my $unwrapped = $option;
+    $unwrapped =~ s/^B<//;
+    $unwrapped =~ s/>$//;
+    # $unmarked is the option with *all* markup removed
+    (my $unmarked = $option) =~ s/[BIL]<|>//msg;
+
+    # If we found an option, check it, collect it
+    if ($unwrapped =~ /^\s*-/) {
+        if ($unmarked =~ /^-${option_re}$/) {
+            return $unwrapped;
+        } else {
+            return "";          # Malformed option
+        }
+    }
+    return undef;               # Something else
+}
+
+sub option_check {
+    my ($id, $filename, $contents) = @_;
+    my $synopsis = ($contents =~ /=head1\s+SYNOPSIS(.*?)=head1/s, $1);
+
+    # Some pages have more than one OPTIONS section, let's make sure
+    # to get them all
+    my $options = '';
+    while($contents =~ /=head1\s+[A-Z ]*?OPTIONS$(.*?)(?==head1)/msg) {
+        $options .= $1;
+    }
+
+    my %synopsis_opts = ();
+    my %options_opts = ();
+
+
+    # Look for options with no or incorrect markup
+    while ($synopsis =~
```

Comment:
same comment about paren/spacing.

Reply:
Wait what, *inside* the parents too?

Reply:
no.  "while space paren space expression space paren"  look at any other "while (" or "if (" in this script for examples.

Reply:
This script isn't consistent in that regard...

Reply:
And why say "no" to then immediately confirm what I asked?

Reply:
I thought you meant sub-expressions within the overall expression.
If the script isn't consistent, I would consider that a bug.  I guess there are some:
```
; grep 'if ([^ ]' find-doc-nits
            if ((-f "$dirname/$n.pod.in" || -f "$dirname/$n.pod")
    if ($opt_o) {
    } elsif ($opt_v) {
    my @missing = loadmissing($missingfile) if ($opt_v);
        if ($name =~ /\s/) {
        if (! exists $name_collection{$name_sec}) {
        } elsif ($filename eq $name_collection{$name_sec}) {
    if ($opt_o) {
; grep 'while ([^ ]' find-doc-nits
        exit 1
;  
```